src/components/MessageInput/hooks/useMessageDetailsForState.ts

```diff
@@ -0,0 +1,123 @@
+import { useEffect, useState } from 'react';
+
+import type { Attachment, UserResponse } from 'stream-chat';
+
+import { generateRandomId } from '../utils/generateRandomId';
+
+import {
+  isEditingBoolean,
+  MessagesContextValue,
+} from '../../../contexts/messagesContext/MessagesContext';
+
+import type {
+  DefaultAttachmentType,
+  DefaultChannelType,
+  DefaultCommandType,
+  DefaultEventType,
+  DefaultMessageType,
+  DefaultReactionType,
+  DefaultUserType,
+} from '../../../types/types';
+
+export type FileUpload = {
+  file: {
+    name: string;
+    size?: number | string;
+    type?: string;
+    uri?: string;
+  };
+  id: string;
+  state: string;
+  url?: string;
+};
+
+export type ImageUpload = {
+  file: {
+    name?: string;
+    uri?: string;
+  };
+  id: string;
+  state: string;
+  url?: string;
+};
+
+export const useMessageDetailsForState = <
+  At extends DefaultAttachmentType = DefaultAttachmentType,
+  Ch extends Record<string, unknown> = DefaultChannelType,
+  Co extends string = DefaultCommandType,
+  Ev extends Record<string, unknown> = DefaultEventType,
+  Me extends Record<string, unknown> = DefaultMessageType,
+  Re extends Record<string, unknown> = DefaultReactionType,
+  Us extends Record<string, unknown> = DefaultUserType
+>(
+  message: MessagesContextValue<At, Ch, Co, Ev, Me, Re, Us>['editing'],
+  initialValue?: string,
+) => {
+  const [fileUploads, setFileUploads] = useState<FileUpload[]>([]);
+  const [imageUploads, setImageUploads] = useState<ImageUpload[]>([]);
+  const [mentionedUsers, setMentionedUsers] = useState(
+    !isEditingBoolean<At, Ch, Co, Ev, Me, Re, Us>(message) &&
+      message.mentioned_users
+      ? [
+          ...(message.mentioned_users as UserResponse<Us>[]).map(
```

Comment:
This is the Message vs MessageResponse issue yeah?

Reply:
yeah